### PR TITLE
Tracking version

### PR DESCRIPTION
**Pull request by gera-gas**
**New version check.**
This pull request introduces an always-opt-in solution for checking a new Audacity version and further download updates by user actions. This feature will be only enabled if  `audacity_has_updates_check` and `audacity_has_networking` is on.

Dialog example:

### CODE REVIEW

File: src/update/VersionId.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <wx/arrstr.h>
+
+// Initialized of current build version by default.
```
Comment (crsib) [inline src/update/VersionId.h:5]: No longer applies

**Review by crsib — CHANGES_REQUESTED** [review]
There are several places, where documentation and a file opening comment are missing. Please add those.

Also, the case that we have multiple Audacity windows opened should be handled.

**Comment by crsib** [comment]
> I disagree that the case of multiple Audacity windows needs to be handled. This dialog is only meant to happen once when the program begins. The UpdateManager doesn't really depend on the project, except to find a main window.

Originally, there was a problem:
1. User opens a window, gets a notification, dismisses it.
2. User opens another project in a new window and gets the notification again.

I asked Anton to fix it by checking the last time we've shown the update dialog. I haven't looked at #966 yet though

**Comment by Paul-Licameli** [comment]
Tl;dr: The updater is a long-lived singleton object, with a long period timer.  Projects and windows have shorter lives.  The updater should not be permanently associated with any one project or window.  On Mac There might even be none at the time the updater needs to notify you.
